### ISSUE TITLE
Define map only for chain_type of RetrievalQA

### ISSUE THREAD

**Issue by Matthieu-Tinycoaching** [langchain-ai/langchain#5935]
### Feature request

Hi,

I would like to use `RetrievalQA` chain with `chain_type` equal to map only. Is it actually possible with LangChain library? When generating answer I also get logits scores for each answer, so I would be able to choose the best answer based on these logits scores. Consequently, I only need the `RetrievalQA` chain to generate answer for each document retrieved. 

### Motivation

Logits scores is a new feature of hugging Face ecosystem for generative model. It would be interesting to enable the user to generate answer for each retrieved document then let the user decided the best answer based on these scores. So a `chain_type` of type map only is needed.

### Your contribution

Let me know if this exists, otherwise I would be glad to test.
